shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -888,16 +895,21 @@ - (instancetype)initWithTarget:(id)target
 }
 
 - (void)touchesBegan:(NSSet*)touches withEvent:(UIEvent*)event {
-  [_platformViewsController->getFlutterViewController() touchesBegan:touches withEvent:event];
+  if (_currentTouchPointersCount == 0) {
+    // At the start of each gesture sequence, we reset the `_flutterViewController`,
+    // so that all the touch events in the same sequence are forwarded to the same `_flutterViewController`.
+    _flutterViewController = _platformViewsController->getFlutterViewController();
+  }
+  [_flutterViewController touchesBegan:touches withEvent:event];
```

Comment:
This still makes me a bit nervous.  Particularly the fact that _flutterViewController is a weak reference and we are sending messages to it potentially after _platformViewsController has gotten rid of it.  My suggestion was to keep a map between events and FlutterViewControllers.

```objc
- (void)touchesBegan:(NSSet*)touches withEvent:(UIEvent*)event {
  _flutterViewControllers[event] = _platformViewsController->getFlutterViewController();
  [_flutterViewControllers[event] touchesBegan:touches withEvent:event];
}

-(void)touchesEnded:(NSSet*)touches withEvent:(UIEvent*)event {
  [_flutterViewControllers[event] touchesEnded:touches withEvent:event];
  [_flutterViewControllers removeObject:event];
}
```

This has the added benefit of having a retain on the flutter view controller while it is being interacted with and everyone gets the correct message.  You'll have to verify that the same event is getting passed across all of the calls.